contrib/platform/src/com/sun/jna/platform/win32/IPHlpAPI.java

```diff
@@ -264,6 +309,121 @@ public FIXED_INFO() {
         public int dwNumAddrs;
     }
 
+    /**
+     * Contains information that describes an IPv4 TCP connection.
+     */
+    @FieldOrder({ "dwState", "dwLocalAddr", "dwLocalPort", "dwRemoteAddr", "dwRemotePort", "dwOwningPid" })
+    class MIB_TCPROW_OWNER_PID extends Structure {
+        public int dwState;
+        public int dwLocalAddr;
+        public int dwLocalPort;
+        public int dwRemoteAddr;
+        public int dwRemotePort;
+        public int dwOwningPid;
+    }
+
+    /**
+     * Contains a table of IPv4 TCP connections on the local computer.
+     */
+    @FieldOrder({ "dwNumEntries", "table" })
+    class MIB_TCPTABLE_OWNER_PID extends Structure {
+        public int dwNumEntries;
+        public MIB_TCPROW_OWNER_PID[] table = new MIB_TCPROW_OWNER_PID[1];
+
+        public MIB_TCPTABLE_OWNER_PID(Memory buf) {
+            super(buf);
+            // First element contains array size
+            table = (MIB_TCPROW_OWNER_PID[]) new MIB_TCPROW_OWNER_PID().toArray(buf.getInt(0));
+            read();
```

Comment:
Could you hit the special case dwNumEntries = 0? If so, I would special case it here and either set table to null or to the empty string and not call into super#read.

Reply:
Good catch.  Null is an option, but it's an array of structures. When you said empty string did you mean empty array?